jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/ControllerHandler.java

```diff
@@ -30,4 +30,6 @@
   PodTemplateSpec getPodTemplate(T controller);
 
   void overrideReplicas(KubernetesListBuilder resources, int replicas);
+
+  void overrideImagePullPolicy(KubernetesListBuilder resources, String imagePullPolicy);
```

Comment:
I think you might be able to provide a default behavior by leveraging a Container visitor.